src/wp-includes/block-supports/block-style-variations.php

```diff
@@ -19,6 +21,7 @@
  * @return string The unique variation name.
  */
 function wp_create_block_style_variation_instance_name( $block, $variation ) {
+	_deprecated_function( __FUNCTION__, '6.7.0', 'wp_unique_id' );
 	return $variation . '--' . md5( serialize( $block ) );
```

Comment:
Is there anything preventing using the replacement?

```suggestion
	return wp_unique_id( "$variation--" );
```

Reply:
Internally, I don't think so. 

The class was being matched on using a regex. So if this was being deprecated due to potential 3rd party use, I leant toward keeping it as it was.

If we change it here, is there much point in deprecating the function?

Reply:
> The class was being matched on using a regex. So if this was being deprecated due to potential 3rd party use, I leant toward keeping it as it was.

Thanks, that makes sense.